### PR TITLE
Get sunflower art

### PR DESCRIPTION
**Pull request by percworld**
## References 
* Links available in issue: [Promises, es6, react Hooks!](https://github.com/mattumland/Salon/projects/1#card-58865214)
* Link to the source:  [JavaScript Docs on Promises](https://javascript.info/async) 

#### Data owned by State: - Object Data Layers - Searched ID’s -
State changes on Pageload: New Users (no favorites on local storage)  -generate search terms randomly  -fetch from search endpoint   -(return is an array of ID’s of objects matching criteria  -List of Matching ID’s saves to state.Existing users(favorites in local storage)-Favorites in state updates 
Route to Learning together: Period of independent study, Show and tell Standup/ Sharing of interesting finds and resources

### Period of time to ingest these shared ideas before attempting to discuss how they will apply in our instance:
.5 Days.

### Period of time to ingest these shared ideas before attempting to implement them in our Components:
1 Day.

### CODE REVIEW

File: src/utilities.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  //   console.log(acc)
+  //   // if (!acc.includes(index)) {
+  //   acc.push(Math.floor(Math.random() * arr.length))
+  //   // }
+  // }, [])
+  // return idList;
+
+  // using Array sort and Math.random
+
+  return array.sort(() => 0.5 - Math.random());
```
Comment (mattumland) [inline src/utilities.js:16]: Dig this cool function

**Review by mattumland — APPROVED** [review]

File: src/components/App/App.js [hunk 2]
```diff
@@ -37,18 +35,26 @@ function App() {
   }
 
   useEffect(() => {
-    getIDs()
+    getIDs();
   }, [])
 
   useEffect(() => {
-    ids && getSingleArtPiece();
+    ids.length && getSingleArtPiece(shuffleItems(ids)[0]);
```
Comment (rdtho2525) [inline src/components/App/App.js:42]: Opportunity for refactoring; we could consider storing the invocation of `shuffleItems` in a variable, then test one condition, and pass in the variable and appropriate index as an argument to `getSingleArtPiece`. i.e. - 
``` 
const shuffledIDs = shuffleItems(ids);

{ids.length && (
  getSingleArtPiece(shuffledIDs[0])
  getSingleArtPiece(shuffledIDs[0])
  etc...
)}
```

File: src/components/App/App.js [hunk 3]
```diff
@@ -37,18 +35,26 @@ function App() {
...
+    ids.length && getSingleArtPiece(shuffleItems(ids)[3]);
+    ids.length && getSingleArtPiece(shuffleItems(ids)[4]);
   }, [ids])
 
   return (
-    <div className="App">
-      {/* {ids.length && console.log('IDs: ', ids)} */}
-      {/* {wall.length && console.log('WALL: ', wall)} */}
-      {wall.length > 1 && <img src={wall[1].primaryImageSmall} />}
+    <div className="App">Salon
```
Comment (rdtho2525) [inline src/components/App/App.js:50]: Opportunity for refactoring; perhaps we could test one condition to render all elements.

File: src/components/App/App.js (hunk 2 above)
Comment (mattumland) [reply src/components/App/App.js:42]: Yeah, that's a good call.

**Comment by rdtho2525** [comment]
Posted a couple of comments regarding opportunities for refactoring and possible solutions. Let me know if these are useful or if there are other recommendations.

File: src/components/App/App.js (hunk 2 above)
Comment (percworld) [reply src/components/App/App.js:42]: I think this is just getting something on the wall as a test and that shuffled items will live in state.  There will be one shuffle, most likely in utility file.   Agreed!

**Comment by rdtho2525** [comment]
These opportunities will be retained for a later refactoring branch.  As discussed amongst the three of us, this is approved to merge.

**Pull request merged**